velero/datadog_checks/velero/metrics.py

```diff
@@ -0,0 +1,85 @@
+# (C) Datadog, Inc. 2025-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+METRIC_MAP = {
+    'go_gc_duration_seconds': 'go.gc_duration_seconds',
+    'go_gc_duration_seconds_sum': 'go.gc_duration_seconds.sum',
+    'go_gc_duration_seconds_count': 'go.gc_duration_seconds',
+    'go_goroutines': 'go.goroutines',
+    'go_info': 'go.info',
+    'go_memstats_alloc_bytes': {'name': 'go.memstats.alloc_bytes', 'type': 'native_dynamic'},
+    'go_memstats_buck_hash_sys_bytes': 'go.memstats.buck_hash_sys_bytes',
+    'go_memstats_frees': 'go.memstats.frees',
+    'go_memstats_gc_sys_bytes': 'go.memstats.gc_sys_bytes',
+    'go_memstats_heap_alloc_bytes': 'go.memstats.heap.alloc_bytes',
+    'go_memstats_heap_idle_bytes': 'go.memstats.heap.idle_bytes',
+    'go_memstats_heap_inuse_bytes': 'go.memstats.heap.inuse_bytes',
+    'go_memstats_heap_objects': 'go.memstats.heap.objects',
+    'go_memstats_heap_released_bytes': 'go.memstats.heap.released_bytes',
+    'go_memstats_heap_sys_bytes': 'go.memstats.heap.sys_bytes',
+    'go_memstats_last_gc_time_seconds': 'go.memstats.last_gc_time_seconds',
+    'go_memstats_lookups': 'go.memstats.lookups',
+    'go_memstats_mallocs': 'go.memstats.mallocs',
+    'go_memstats_mcache_inuse_bytes': 'go.memstats.mcache.inuse_bytes',
+    'go_memstats_mcache_sys_bytes': 'go.memstats.mcache.sys_bytes',
+    'go_memstats_mspan_inuse_bytes': 'go.memstats.mspan.inuse_bytes',
+    'go_memstats_mspan_sys_bytes': 'go.memstats.mspan.sys_bytes',
+    'go_memstats_next_gc_bytes': 'go.memstats.next_gc_bytes',
+    'go_memstats_other_sys_bytes': 'go.memstats.other_sys_bytes',
+    'go_memstats_stack_inuse_bytes': 'go.memstats.stack.inuse_bytes',
+    'go_memstats_stack_sys_bytes': 'go.memstats.stack.sys_bytes',
+    'go_memstats_sys_bytes': 'go.memstats.sys_bytes',
+    'go_threads': 'go.threads',
+    'podVolume_data_download_cancel': 'podVolume.data.download.cancel',
+    'podVolume_data_download_failure': 'podVolume.data.download.failure',
+    'podVolume_data_download_success': 'podVolume.data.download.success',
+    'podVolume_data_upload_cancel': 'podVolume.data.upload.cancel',
+    'podVolume_data_upload_failure': 'podVolume.data.upload.failure',
+    'podVolume_data_upload_success': 'podVolume.data.upload.success',
+    'podVolume_pod_volume_backup_dequeue_count': 'podVolume.pod_volume.backup.dequeue',
+    'podVolume_pod_volume_backup_enqueue_count': 'podVolume.pod_volume.backup.enqueue',
+    'podVolume_pod_volume_operation_latency_seconds': 'podVolume.pod_volume.operation_latency_seconds',
+    'podVolume_pod_volume_operation_latency_seconds_gauge': 'podVolume.pod_volume.operation_latency_seconds.gauge',
+    'process_cpu_seconds': 'process.cpu_seconds',
+    'process_max_fds': 'process.max_fds',
+    'process_open_fds': 'process.open_fds',
+    'process_resident_memory_bytes': 'process.resident_memory_bytes',
+    'process_start_time_seconds': 'process.start_time_seconds',
```

Comment:
heheh time_elapsed metric type.

Reply:
Non blocking nit:

Maybe we can also start QAing and using this:
https://github.com/DataDog/integrations-core/blob/f2d5a9be624b0eeed073c54c0717da78dea83dc7/datadog_checks_base/datadog_checks/base/checks/openmetrics/v2/metrics.py#L4

The idea was that we'd just import this as the Go default metrics, so we don't have to always worry about the naming conventions. But the list there seems a bit short. Maybe we should update it going forward with new entries that aren't on there.

Reply:
➕ to using the go default metrics! iirc I commented on this in the RFC as well.